provision/provision-site.sh

```diff
@@ -30,6 +30,26 @@ VVV_CONFIG=/vagrant/config.yml
 
 . "/srv/provision/provisioners.sh"
 
+# @description Retrieves a config value for the given site as specified in `config.yml`
+#
+# @arg $1 string the config value to fetch
+# @arg $2 string the default value
+function vvv_get_site_config_value() {
+  local value=$(shyaml -q get-value "sites.${SITE_ESCAPED}.${1}" "${2}" < ${VVV_CONFIG})
+  echo "${value}"
+}
+php_version=$(readlink -f /usr/bin/php)
+if [[ $php_version != *"${VVV_BASE_PHPVERSION}"* ]]; then
+  DEFAULTPHP=$(vvv_get_site_config_value 'php' "${VVV_BASE_PHPVERSION}")
+  echo " * Setting the default PHP CLI version ( ${DEFAULTPHP} ) for this site"
+  noroot update-alternatives --set php "/usr/bin/php${DEFAULTPHP}"
+  noroot update-alternatives --set phar "/usr/bin/phar${DEFAULTPHP}"
+  noroot update-alternatives --set phar.phar "/usr/bin/phar.phar${DEFAULTPHP}"
+  noroot update-alternatives --set phpize "/usr/bin/phpize${DEFAULTPHP}"
+  noroot update-alternatives --set php-config "/usr/bin/php-config${DEFAULTPHP}"
```

Comment:
does `noroot` make sense here? We run these commands a lot, maybe it's time to turn them into functions, perhaps:

`vvv_set_php_cli_version` and `vvv_reset_php_cli_version`, where the following behaviour happens:

```sh
vvv_set_php_cli_version "7.2" # php is now at 7.2
vvv_reset_php_cli_version # php is now at the default 7.4
vvv_set_php_cli_version  # no parameter so it calls vvv_reset_php_cli_version
```

Reply:
I preferred to keep the function as you suggest but read the parameter by the config instead as parameter

Reply:
the concern is more that broken provisioning might break the expectations here causing more broken provisioning which just sounds painful to debug :(